modules/light-clients/07-tendermint/types/misbehaviour_handle.go

```diff
@@ -58,37 +58,8 @@ func (cs ClientState) CheckMisbehaviourAndUpdateState(
 		}
 	}
 
-	// Regardless of the type of misbehaviour, ensure that both headers are valid and would have been accepted by light-client
-
-	// Retrieve trusted consensus states for each Header in misbehaviour
-	// and unmarshal from clientStore
-
-	// Get consensus bytes from clientStore
-	tmConsensusState1, err := GetConsensusState(clientStore, cdc, tmMisbehaviour.Header1.TrustedHeight)
-	if err != nil {
-		return nil, sdkerrors.Wrapf(err, "could not get trusted consensus state from clientStore for Header1 at TrustedHeight: %s", tmMisbehaviour.Header1)
-	}
-
-	// Get consensus bytes from clientStore
-	tmConsensusState2, err := GetConsensusState(clientStore, cdc, tmMisbehaviour.Header2.TrustedHeight)
-	if err != nil {
-		return nil, sdkerrors.Wrapf(err, "could not get trusted consensus state from clientStore for Header2 at TrustedHeight: %s", tmMisbehaviour.Header2)
-	}
-
-	// Check the validity of the two conflicting headers against their respective
-	// trusted consensus states
-	// NOTE: header height and commitment root assertions are checked in
-	// misbehaviour.ValidateBasic by the client keeper and msg.ValidateBasic
-	// by the base application.
-	if err := checkMisbehaviourHeader(
-		&cs, tmConsensusState1, tmMisbehaviour.Header1, ctx.BlockTime(),
-	); err != nil {
-		return nil, sdkerrors.Wrap(err, "verifying Header1 in Misbehaviour failed")
-	}
-	if err := checkMisbehaviourHeader(
-		&cs, tmConsensusState2, tmMisbehaviour.Header2, ctx.BlockTime(),
-	); err != nil {
-		return nil, sdkerrors.Wrap(err, "verifying Header2 in Misbehaviour failed")
+	if err := cs.ValidateClientMessage(ctx, clientStore, cdc, nil, misbehaviour, ctx.BlockTime()); err != nil {
```

Comment:
@colin-axner Decided to go with `ValidateClientMessage` instead of `VerifyClientMessage` as per a suggestion from @damiannolan . wdyt?

Reply:
Yeah I'm still unsure of the naming. I suggested using `ValidateClientMessage` instead of `VerifyClientMessage` to have some separation between this other `VerifyXyz` methods on the `ClientState` interface which do proof verification.

Reply:
It should be `VerifyClientMessage`. This function is doing proof verification. `Validate` usage in SDK modules implies basic validation

Reply:
Ah yes okay, so a general rule of thumb should be "validate" is used in terms of stateless checks

Reply:
Yes exactly